nntrainer/layers/cl_layers/transpose_cl.cpp

```diff
@@ -0,0 +1,357 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 Niket Agarwal <niket.a@samsung.com>
+ *
+ * @file   transpose_cl.cpp
+ * @date   31 July 2024
+ * @brief  Implementation of transpose layer
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Niket Agarwal <niket.a@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+
+#include "transpose_cl.h"
+#include <iostream>
+#include <layer_context.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <node_exporter.h>
+
+std::string transpose_cl_kernel_fp16_ =
+  R"(
+    #pragma OPENCL EXTENSION cl_khr_fp16 : enable
+    __kernel void transpose_cl_fp16(__global const half* in, 
```

Comment:
If you look at other functions, they don't use n, c, h, w; instead, they use accurate names. How about modifying this function as below?
I believe it is a good idea to use similar naming design even though this code is a GPU kernel code.


For example, like the function below:
```cpp
void TransposeLayerCl::transpose_cl(const float *in, float *res,
                                    unsigned int input_batch_size,
                                    unsigned int input_channels,
                                    unsigned int input_height,
                                    unsigned int input_width,
                                    RunLayerContext &context)
```